src/osversion/test/VersionTranslator_GTest.cpp

```diff
@@ -2548,3 +2553,892 @@ TEST_F(OSVersionFixture, update_3_6_1_to_3_7_0_GroundHeatExchangerVertical) {
+  }
+
+  {
+    std::vector<WorkspaceObject> coils = model->getObjectsByType("OS:Coil:Heating:DX:VariableSpeed");
+    ASSERT_EQ(1u, coils.size());
+    WorkspaceObject coil = coils[0];
+
+    EXPECT_EQ(20, coil.numFields());
+
+    const size_t insertionIndex = 13;
+    EXPECT_EQ(100.0, coil.getDouble(insertionIndex - 1).get());
+
+    // Crankcase Heater Capacity Function of Temperature Curve Name
+    EXPECT_TRUE(coil.isEmpty(insertionIndex));
+
+    EXPECT_EQ(11.0, coil.getDouble(insertionIndex + 1).get());
+
+    // Last field: Speed Data List
+    ASSERT_TRUE(coil.getTarget(19));
+    EXPECT_EQ("CoilHeatingDXVariableSpeed Speed Data List", coil.getTarget(19)->nameString());
+  }
+
+  {
+    std::vector<WorkspaceObject> coils = model->getObjectsByType("OS:Coil:WaterHeating:AirToWaterHeatPump");
+    ASSERT_EQ(1u, coils.size());
+    WorkspaceObject coil = coils[0];
+
+    EXPECT_EQ(30, coil.numFields());
+
+    const size_t insertionIndex = 20;
+    EXPECT_EQ(100.0, coil.getDouble(insertionIndex - 1).get());
+
+    // Crankcase Heater Capacity Function of Temperature Curve Name
+    EXPECT_TRUE(coil.isEmpty(insertionIndex));
+
+    EXPECT_EQ(11.0, coil.getDouble(insertionIndex + 1).get());
+
+    // Last field: Part Load Fraction Correlation Curve
+    ASSERT_TRUE(coil.getTarget(29));
+    EXPECT_EQ("CoilWaterHeatingAirToWaterHeatPump PLFCorrelationCurve", coil.getTarget(29)->nameString());
+  }
+
+  {
+    std::vector<WorkspaceObject> coils = model->getObjectsByType("OS:Coil:WaterHeating:AirToWaterHeatPump:VariableSpeed");
+    ASSERT_EQ(1u, coils.size());
+    WorkspaceObject coil = coils[0];
+
+    EXPECT_EQ(23, coil.numFields());
+
+    const size_t insertionIndex = 18;
+    EXPECT_EQ(100.0, coil.getDouble(insertionIndex - 1).get());
+
+    // Crankcase Heater Capacity Function of Temperature Curve Name
+    EXPECT_TRUE(coil.isEmpty(insertionIndex));
+
+    EXPECT_EQ(11.0, coil.getDouble(insertionIndex + 1).get());
+
+    // Last field: Speed Data List
+    ASSERT_TRUE(coil.getTarget(22));
+    EXPECT_EQ("CoilWaterHeatingAirToWaterHeatPumpVariableSpeed Speed Data List", coil.getTarget(22)->nameString());
+  }
+
+  {
+    std::vector<WorkspaceObject> coils = model->getObjectsByType("OS:Coil:WaterHeating:AirToWaterHeatPump:Wrapped");
+    ASSERT_EQ(1u, coils.size());
+    WorkspaceObject coil = coils[0];
+
+    EXPECT_EQ(21, coil.numFields());
+
+    const size_t insertionIndex = 13;
+    EXPECT_EQ(100.0, coil.getDouble(insertionIndex - 1).get());
+
+    // Crankcase Heater Capacity Function of Temperature Curve Name
+    EXPECT_TRUE(coil.isEmpty(insertionIndex));
+
+    EXPECT_EQ(11.0, coil.getDouble(insertionIndex + 1).get());
+
+    // Last field: Part Load Fraction Correlation Curve Name
+    ASSERT_TRUE(coil.getTarget(20));
+    EXPECT_EQ("CoilWaterHeatingAirToWaterHeatPumpWrapped PLFCorrelationCurve", coil.getTarget(20)->nameString());
+  }
+}
+
+TEST_F(OSVersionFixture, update_3_6_1_to_3_7_0_Coils_Latent_solo) {
```

Comment:
Starting here you have 600 lines of testing for the Coil Latent Stuff, testing all combinations of Unitary/Child coils I could think off. Making sure the PLF Curves are created properly, shared between HC and CC if appropriate, etc.